paddle/fluid/distributed/auto_parallel/spmd_rules/softmax_spmd_rule.cc

```diff
@@ -107,9 +107,71 @@ SoftmaxSPMDRule::InferForward(const std::vector<DistTensorSpec>& input_specs,
 
 std::pair<std::vector<TensorDistAttr>, std::vector<TensorDistAttr>>
 SoftmaxSPMDRule::InferBackward(const std::vector<DistTensorSpec>& input_specs,
+                               const std::vector<DistTensorSpec>& output_specs,
                                const paddle::framework::AttributeMap& attrs) {
-  PADDLE_THROW(phi::errors::Unimplemented(
-      "InferBackward of SoftmaxSPMDRule is NOT implemented yet."));
+  // step0: verify input args based on softmax logic
+  int64_t input_specs_size = input_specs.size();
+  int64_t output_specs_size = output_specs.size();
+  PADDLE_ENFORCE_EQ(
+      input_specs_size,
+      1,
+      phi::errors::InvalidArgument(
+          "The size of InputSpec of softmax should be 1, but got [%d].",
+          input_specs_size));
+  PADDLE_ENFORCE_EQ(
+      output_specs_size,
+      1,
+      phi::errors::InvalidArgument(
+          "The size of InputSpec of softmax should be 1, but got [%d].",
+          output_specs_size));
+  VerifySpecs(output_specs, "softmax_backward");
+
+  // step1: build Einsum Notation
+  std::vector<int64_t> x_shape = input_specs[0].shape();
+  int64_t x_ndim = input_specs[0].shape().size();
+
+  std::string alphabet = "abcdefghijklmnopqrstuvwxyz";
+  std::string x_axes = GetBroadcastAxes(x_ndim, x_ndim, alphabet);
+  std::string out_axes = x_axes;
+
+  int axis = ExtractAttr<int>("axis", attrs);
+  // normalize axis
+  if (axis < 0) {
+    axis = x_ndim + axis;
+  }
+
+  // sharding on softmax_axis is not supported now, so set
+  // the notation on softmax_axis to '1' so that we can set
+  // its dim mapping to -1
+  x_axes[axis] = '1';
```

Comment:
change the einsum_string to "1" to ensure the unsharded maybe not a good idea here, since it would disconnect the relation of softmax_axis between input and output when print the vlog for debug: 

"ab1" --> "abc", later developer might not realize that  axis "1" and "c" are the same axis in two tensors.

it would be better that:
first: out_dims_mapping[axis] = -1;
then: keep the input_axes unchanged.